src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,128 @@
+package duke;
+
+import duke.dukeExceptions.DukeException;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+/**
+ * Represents a Parser class that is responsible for parsing user inputs.
+ * 
+ * @author Ramanathan Kumarappan
+ */
+public class Parser {
+    /**
+     * Parses the given string and executes the appropriate action.
+     * 
+     * @param input - The user command to be parsed.
+     * @param tl - The TaskList associated with the user.
+     * @return - The resultant String of executing the command.
+     * @throws DukeException - When there is an error with the command.
+     */
+    public String parseInput(String input, TaskList tl) throws DukeException {
```

Comment:
I don't think I like tl as the variable name of a TaskList instance. I understand that you have derived it from the abbreviation of TaskList, however, considering the length of the code, tl may be a little too succinct.